client/src/components/PostForm.tsx

```diff
@@ -40,19 +41,27 @@ export const PostForm = (props: {
 }) => {
   const IS_EDITING = props.post !== undefined;
   const AuthToken = localStorage.getItem("BEARER_TOKEN") || "";
-
+  const photoArray = ((props.post?.postPhotos.length || 0) < 1 ? [] : props.post?.postPhotos) || [];
```

Comment:
`props.post?.postPhotos || []`

Also I think you might have to deep clone the array, which should be 

`props.post ? structuredClone(props.post.postPhotos) : []`